pinot-broker/src/main/java/com/linkedin/pinot/broker/routing/builder/BalancedRandomRoutingTableBuilder.java

```diff
@@ -39,41 +39,34 @@
   private int _numRoutingTables = DEFAULT_NUM_ROUTING_TABLES;
 
   @Override
-  public void init(Configuration configuration, TableConfig tableConfig, ZkHelixPropertyStore<ZNRecord> propertyStore, BrokerMetrics brokerMetrics) {
+  public void init(Configuration configuration, TableConfig tableConfig, ZkHelixPropertyStore<ZNRecord> propertyStore,
+      BrokerMetrics brokerMetrics) {
     super.init(configuration, tableConfig, propertyStore, brokerMetrics);
     _numRoutingTables = configuration.getInt(NUM_ROUTING_TABLES_KEY, DEFAULT_NUM_ROUTING_TABLES);
   }
 
-  @Override
-  public void computeRoutingTableFromExternalView(String tableName, ExternalView externalView,
+  Map<String, List<String>> computeSegmentToServersMapOnExternalViewChange(ExternalView externalView,
       List<InstanceConfig> instanceConfigs) {
+    return RoutingTableBuilderUtil.computeSegmentToServersMapForOfflineTable(externalView, instanceConfigs);
```

Comment:
This part should handle the no serving host issue.

Reply:
fixed